### DISCUSSION TITLE
[Question] Removing a namespace from a C# exe

### DISCUSSION THREAD

**Discussion by Th3Hunger** [Washi1337/AsmResolver#345] [General]
### Description

How to remove a whole namespace directly from a C# exe?

**Comment by Washi1337** [comment]
The namespace of a type can be changed via the `Namespace` property ot a `TypeDefinition`. You can simply set it to `null` to remove it from a namespace. e.g.:

```csharp
var module = ModuleDefinition.FromFile("file.exe");

foreach (var type in module.TopLevelTypes)
{
   if (type.Namespace == "<your namespace to remove>")
       type.Namespace = null;
}
```
